packages/domain/src/use-cases/publications/CreatePost.ts

```diff
@@ -8,69 +8,116 @@ import {
+   */
   reference: ReferencePolicyConfig;
-
   /**
-   * The content of a publication. If this is blank then `media` must be defined
+   * A list of tags for the publication. This can be used to categorize the publication.
+   *
+   * These are not the same as #hashtag in the publication content. Use these if you don't want to clutter the publication content with tags.
    */
-  content?: string;
+  tags?: string[];
+};
+
+export type CreateTextualPostRequest = BasePostRequest & {
   /**
-   * Main content focus for this publication
+   * The publication focus is the to text only
    */
-  contentFocus: ContentFocus;
+  contentFocus: ContentFocus.ARTICLE | ContentFocus.LINK | ContentFocus.TEXT_ONLY;
   /**
-   * IOS 639-1 language code aka en or it and ISO 3166-1 alpha-2 region code
-   * aka US or IT aka en-US or it-IT
+   * The publication content as Markdown string.
    */
-  locale: Locale;
+  content: string;
   /**
-   * Lens supported attached media items to the publication
+   * The publication media. An array of media objects.
    */
   media?: MediaObject[];
+};
+
+export type CreateMediaPostRequest = BasePostRequest & {
   /**
-   * A URL to a multi-media attachment for the item. The file extensions GLTF, GLB, WEBM, MP4, M4V, OGV,
-   * and OGG are supported, along with the audio-only extensions MP3, WAV, and OGA.
-   * Animation_url also supports HTML pages, allowing you to build rich experiences and interactive NFTs using JavaScript canvas,
-   * WebGL, and more. Scripts and relative paths within the HTML page are now supported. However, access to browser extensions is not supported.
+   * The publication content as Markdown string.
    */
-  animationUrl?: Url;
+  content?: string;
   /**
-   * This is the appId the content belongs to
+   * The publication content focus. Determines what is the primary objective of the publication.
    */
-  appId?: AppId;
+  contentFocus: ContentFocus.AUDIO | ContentFocus.IMAGE | ContentFocus.VIDEO;
   /**
-   * Ability to add a content warning
+   * The publication media. An array of media objects.
    */
-  contentWarning?: ContentWarning;
+  media: MediaObject[];
+};
+
+export type CreateEmbedPostRequest = BasePostRequest & {
   /**
-   * This is the URL that will appear below the asset's image on OpenSea and others
-   * and will allow users to leave OpenSea and view the item on the site.
+   * A URL to a multi-media attachment for the item. The file extensions GLTF, GLB, WEBM, MP4, M4V, OGV,
+   * and OGG are supported, along with the audio-only extensions MP3, WAV, and OGA.
+   * Animation_url also supports HTML pages, allowing you to build rich experiences and interactive NFTs using JavaScript canvas,
+   * WebGL, and more. Scripts and relative paths within the HTML page are now supported. However, access to browser extensions is not supported.
    */
```

Comment:
I need to polish this copy, could lead to odd audio post if we keep it as it is.